src/database/migration/file/base_migration_v2.ts

```diff
@@ -0,0 +1,45 @@
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+abstract class BaseMigrationV2 {
+  abstract _migrationVersion: number;
+
+  abstract _tableName: string;
+
+  abstract upQuery(): string;
+
+  abstract downQuery(): string | null;
+
+  async up(db: SQLiteDatabase): Promise<void> {
+    console.log(
+      `====== MIGRATING VERSION ${
+        this._migrationVersion
+      }: ${this._tableName.toLocaleUpperCase()} TABLE ======`
+    );
+
+    await db.executeSql(this.upQuery());
+
+    console.log(
+      `====== DONE MIGRATING VERSION ${
+        this._migrationVersion
+      }: ${this._tableName.toLocaleUpperCase()}  TABLE =====`
+    );
+  }
+
+  async down(db: SQLiteDatabase): Promise<void> {
+    console.log(
+      `====== DOWNGRADING VERSION ${
+        this._migrationVersion
+      }: ${this._tableName.toLocaleUpperCase()} TABLE ======`
+    );
+    const query = this.downQuery();
+    if (query) await db.executeSql(query);
+
+    console.log(
+      `====== DONE DOWNGRADING VERSION ${
+        this._migrationVersion
+      }: ${this._tableName.toLocaleUpperCase()}  TABLE ======`
+    );
+  }
+}
+
+export default BaseMigrationV2;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but I would suggest removing the console.log statements or replacing them with a more sophisticated logging mechanism. Console.log statements can slow down your application and expose potentially sensitive information in a production environment.
```diff
  async up(db: SQLiteDatabase): Promise<void> {
    // Removed console.log for performance and security reasons

    await db.executeSql(this.upQuery());

    // Removed console.log for performance and security reasons
  }

  async down(db: SQLiteDatabase): Promise<void> {
    // Removed console.log for performance and security reasons

    const query = this.downQuery();
    if (query) await db.executeSql(query);

    // Removed console.log for performance and security reasons
  }
```
If you need to keep logs for debugging purposes, consider using a logging library that can be toggled for development/production environments or provides log levels.


<!-- This is an auto-generated comment by OSS CodeRabbit -->